frontend/packages/console-shared/src/hooks/useUserSettings.ts

```diff
@@ -40,7 +40,7 @@ export const useUserSettings = <T>(
   sync: boolean = false,
 ): [T, React.Dispatch<React.SetStateAction<T>>, boolean] => {
   const impersonate: boolean = useSelector((state: RootState) => !!getImpersonate(state));
-  const keyRef = React.useRef<string>(key);
+  const keyRef = React.useRef<string>(key?.replace(/[^-._a-zA-Z0-9]/g, '_'));
```

Comment:
There are different components that call `useUserSettings` without a key, that's the reason for the optional chaining here. At the moment this works fine but maybe should take a look to improve this later on.